pkg/common/ipsec/utils_test.go

```diff
@@ -5,31 +5,44 @@ package ipsec
 
 import (
 	"encoding/hex"
+	"log"
 	"net"
 	"testing"
 
 	"github.com/stretchr/testify/require"
 	"github.com/vishvananda/netlink"
 )
 
-func getXfrmState(src string, dst string, spi int, key string, mark uint32) netlink.XfrmState {
+func getXfrmState(src string, dst string, spi int, algoName string, key string, mark uint32) netlink.XfrmState {
 	k, _ := hex.DecodeString(key)
-	return netlink.XfrmState{
+	state := netlink.XfrmState{
 		Src:   net.ParseIP(src),
 		Dst:   net.ParseIP(dst),
 		Proto: netlink.XFRM_PROTO_ESP,
 		Mode:  netlink.XFRM_MODE_TUNNEL,
 		Spi:   spi,
-		Aead: &netlink.XfrmStateAlgo{
-			Name:   "rfc4106(gcm(aes))",
-			Key:    k,
-			ICVLen: 64,
-		},
 		Mark: &netlink.XfrmMark{
 			Value: mark,
 			Mask:  0xffffff00,
 		},
 	}
+	switch algoName {
+	case "cbc(aes)":
+		state.Crypt = &netlink.XfrmStateAlgo{
+			Name:   "cbc(aes)",
+			Key:    k,
+			ICVLen: 64,
+		}
+	case "rfc4106(gcm(aes))":
+		state.Aead = &netlink.XfrmStateAlgo{
+			Name:   "rfc4106(gcm(aes))",
+			Key:    k,
+			ICVLen: 64,
+		}
+	default:
+		log.Fatalf("Unsupported algorithm: %s", algoName)
```

Comment:
We should fail the test via the `testing` or `require` primitives in this case, not fatal via `log`.

Reply:
Fixed.